### PR TITLE
Last validators

### PR DESCRIPTION
**Pull request by reekee94**

### CODE REVIEW

File: client/src/components/checkout-form/checkout-form.js [hunk 1]
```diff
@@ -146,9 +150,18 @@ const CheckoutForm = ({
...
 
   const handleChange = (event) => {
+    const erora = universal(event.target.name, event.target.value)
     event.persist();
     setOrder({ ...order, [event.target.name]: event.target.value });
     setOrderToStore(order);
+    setEmailError({...emailError, [event.target.name]: erora})
+    setValidated(true)
+    Object.entries(emailError).forEach(([key, value]) => {
+      console.log(`${key} -- ${value}`);
```
Comment (bandvov) [inline client/src/components/checkout-form/checkout-form.js:160]: please remove console.log()

File: client/src/components/checkout-form/checkout-form.js [hunk 2]
```diff
@@ -168,109 +179,92 @@ const CheckoutForm = ({
                 <Form.Group controlId="firstNameValidate">
                   <Form.Label>Firstname</Form.Label>
                   <Form.Control
-                    required
                     placeholder={placeholder}
                     name={'firstName'}
                     onChange={handleChange}
                     value={order.firstName}
+                    //onBlur = {validationState}
```
Comment (bandvov) [inline client/src/components/checkout-form/checkout-form.js:186]: please remove comment

File: client/src/components/checkout-form/checkout-form.js [hunk 3]
```diff
@@ -168,109 +179,92 @@ const CheckoutForm = ({
...
                 </Form.Group>
                 <Form.Group controlId="lastNameValidate">
                   <Form.Label>Lastname</Form.Label>
                   <Form.Control
-                    required
                     placeholder={placeholder}
                     name={'lastName'}
                     onChange={handleChange}
                     value={order.lastName}
+                    //onBlur = {validationState}
```
Comment (bandvov) [inline client/src/components/checkout-form/checkout-form.js:198]: please remove comment

File: client/src/components/login/Login.js [hunk 4]
```diff
@@ -55,6 +45,14 @@ const Login = ({ storeService, setUserLogged, setUserLoading, userLogged, userLo
     const handleChange = (event) => {
         event.persist();
         setUser(prevUser => ({ ...prevUser, [event.target.name]: event.target.value }));
+        const err = universal(event.target.name, event.target.value)
+        if(event.target.name == 'email'){
+            console.log("here")
```
Comment (bandvov) [inline client/src/components/login/Login.js:50]: please remove console.log()

File: client/src/components/register/Register.js [hunk 5]
```diff
@@ -127,82 +145,84 @@ const Register = ({
...
             value={user.firstName}
             onChange={handleChange}
             placeholder="Enter firstname..."
           />
-
+          <i className="text-danger position-static">{firstNameError}</i>
         </Form.Group>
         <Form.Group>
-          <Form.Label>Last name</Form.Label>
+          <Form.Label>Last name<sup style = {{ color: "red" }}>*</sup></Form.Label>
```
Comment (bandvov) [inline client/src/components/register/Register.js:159]: please move styles to css file

File: client/src/components/register/Register.js [hunk 6]
```diff
@@ -127,82 +145,84 @@ const Register = ({
...
             value={user.lastName}
             onChange={handleChange}
             placeholder="Enter lastname..."
           />
+          <i className="text-danger position-static">{lastNameError}</i>
         </Form.Group>
 
         <Form.Group controlId="formBasicEmail">
-          <Form.Label>Email address</Form.Label>
+          <Form.Label>Email address<sup style = {{ color: "red" }}>*</sup></Form.Label>
```
Comment (bandvov) [inline client/src/components/register/Register.js:171]: please move style to css file

File: client/src/components/register/Register.js [hunk 7]
```diff
@@ -127,82 +145,84 @@ const Register = ({
...
-          <i className="text-danger position-static">{emailErrorMessage}</i>
+          <i className="text-danger position-static">{emailError}</i>
           <Form.Text className="text-muted">
             We'll never share your email with anyone else.
           </Form.Text>
         </Form.Group>
 
         <Form.Group controlId="formBasicPassword">
-          <Form.Label>Password</Form.Label>
+          <Form.Label>Password<sup style = {{ color: "red" }}>*</sup></Form.Label>
```
Comment (bandvov) [inline client/src/components/register/Register.js:187]: please move style to css file

File: client/src/components/register/Register.js [hunk 8]
```diff
@@ -127,82 +145,84 @@ const Register = ({
...
           <Form.Group className="pass-wrapper">
             <Form.Control
               type={passwordShown ? 'text' : 'password'}
               placeholder="Enter password..."
               name={'password'}
               value={user.password}
               onChange={handleChange}
-              title="min length 8 max 16 characters"
+              //onBlur={validateConfirmPassword}
```
Comment (bandvov) [inline client/src/components/register/Register.js:195]: please remove comment

File: client/src/components/register/Register.js [hunk 9]
```diff
@@ -127,82 +145,84 @@ const Register = ({
...
+              onClick={togglePasswordVisiblity}
             />
           </Form.Group>
           <i className="text-danger position-static">
-            {confirmPasswordErrorMessage}
+            {confirmPasswordError}
           </i>
         </Form.Group>
-
+        <Form.Label style = {{ fontSize: "14px" }}> Required field marked by <sup style = {{ color: "red" }}>*</sup></Form.Label>
```
Comment (bandvov) [inline client/src/components/register/Register.js:225]: please move styles to css file

File: client/src/components/checkout-form/checkout-form.js [hunk 10]
```diff
@@ -168,109 +179,92 @@ const CheckoutForm = ({
...
                 <Form.Group controlId="emailValidate">
                   <Form.Label>Email</Form.Label>
                   <Form.Control
-                    required
                     placeholder={placeholder}
                     name={'email'}
                     type="email"
                     onChange={handleChange}
                     value={order.email}
+                    //onBlur = {validationState}
```
Comment (bandvov) [inline client/src/components/checkout-form/checkout-form.js:210]: please remove comment

File: client/src/components/checkout-form/checkout-form.js [hunk 11]
```diff
@@ -168,109 +179,92 @@ const CheckoutForm = ({
...
                 <Form.Group controlId="phoneValidate">
                   <Form.Label>Contact Phone Number</Form.Label>
                   <Form.Control
-                    required
                     type="number"
                     placeholder={placeholder}
                     name="contactPhone"
                     onChange={handleChange}
                     value={order.contactPhone}
+                    //onBlur = {validationState}
```
Comment (bandvov) [inline client/src/components/checkout-form/checkout-form.js:222]: please remove comment

File: client/src/components/checkout-form/checkout-form.js [hunk 12]
```diff
@@ -168,109 +179,92 @@ const CheckoutForm = ({
...
-                    Please type Phone number. This field is required
-                  </Form.Control.Feedback>
+                  <i className="text-danger position-static">{emailError.phone}</i>
                 </Form.Group>
                 <Form.Group>
                   <CheckoutSelect
                     selectOptions={countries}
                     name="country"
                     handleChange={handleChange}
+                    // onBlur = {validationState}
```
Comment (bandvov) [inline client/src/components/checkout-form/checkout-form.js:231]: please remove comment

File: client/src/components/checkout-form/checkout-form.js [hunk 13]
```diff
@@ -168,109 +179,92 @@ const CheckoutForm = ({
...
                 </Form.Group>
                 <Form.Group controlId="cityValidate">
                   <Form.Label>City</Form.Label>
                   <Form.Control
-                    required
                     placeholder={placeholder}
                     name={'city'}
                     onChange={handleChange}
                     value={order.city}
+                    //onBlur = {validationState}
```
Comment (bandvov) [inline client/src/components/checkout-form/checkout-form.js:243]: please remove comment

File: client/src/components/checkout-form/checkout-form.js [hunk 14]
```diff
@@ -168,109 +179,92 @@ const CheckoutForm = ({
...
                 </Form.Group>
                 <Form.Group controlId="streetValidate">
                   <Form.Label required>Street</Form.Label>
                   <Form.Control
-                    required
                     placeholder={placeholder}
                     name="street"
                     onChange={handleChange}
                     value={order.street}
+                    //onBlur = {validationState}
```
Comment (bandvov) [inline client/src/components/checkout-form/checkout-form.js:254]: please remove comment

File: client/src/components/checkout-form/checkout-form.js [hunk 15]
```diff
@@ -168,109 +179,92 @@ const CheckoutForm = ({
...
                 </Form.Group>
                 <Form.Group controlId="buildingValidate">
                   <Form.Label>Building number</Form.Label>
                   <Form.Control
-                    required
                     placeholder={placeholder}
                     name="buildingNumber"
                     onChange={handleChange}
                     value={order.buildingNumber}
+                    //onBlur = {validationState}
```
Comment (bandvov) [inline client/src/components/checkout-form/checkout-form.js:265]: please remove comment

**Review by bandvov — CHANGES_REQUESTED** [review]

File: client/src/components/checkout-form/checkout-form.js [hunk 16]
```diff
@@ -146,9 +150,18 @@ const CheckoutForm = ({
   };
 
   const handleChange = (event) => {
+    const erora = universal(event.target.name, event.target.value)
```
Comment (oleksandrhodzevych) [inline client/src/components/checkout-form/checkout-form.js:153]: errors

**Review by oleksandrhodzevych — CHANGES_REQUESTED** [review]

**Comment by VolodymyrTrach** [comment]
Pull request should be to development branch not to master

**Comment by VolodymyrTrach** [comment]
bad checkout validation

File: client/src/validators/form-validators.js [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+                return 'Please choose a payment method'
+            }
+            break
+        case type === 'deliveryType':
+            if (text.match(formRegExp.deliveryMethod)) {
+                return false
+            } else {
+                return 'Please choose a delivery type'
+            }
+            break
```
Comment (VolodymyrTrach) [inline client/src/validators/form-validators.js:81]: The switch should have default case

Comment (VolodymyrTrach) [inline client/src/validators/form-validators.js:81]: All breaks are Unreachable.

File: client/src/components/register/Register.js [hunk 18]
```diff
@@ -3,21 +3,17 @@ import "./Register.css";
 import { Form, Button, Modal } from "react-bootstrap";
 import { Link, Redirect } from "react-router-dom";
 import { connect } from "react-redux";
+import {universal} from '../../validators/form-validators'
 import { setUserLogged, setUserLoading } from "../../actions";
 import withStoreService from "../hoc";
 import { setToLocalStorage } from "../../services/localStoreService";
-
+import {formRegExp} from '../../configs/frontend-config'
```
Comment (VolodymyrTrach) [inline client/src/components/register/Register.js:10]: formRegExp is never used. 
use it or remove

File: client/src/components/register/Register.js [hunk 19]
```diff
@@ -39,26 +35,46 @@ const Register = ({
...
+    else{
+      setAllFieldsValidated(false)
+    }
+  }, [
+      allFieldsValidated,
+      agreedWithTerms,
+      emailError,
+      confirmPasswordError,
+      lastNameError,
+      firstNameError ])
```
Comment (VolodymyrTrach) [inline client/src/components/register/Register.js:77]: missing dependency: passwordError

**Review by VolodymyrTrach — CHANGES_REQUESTED** [review]
remove all changes from the checkout page

**Pull request closed without merging**